### PR TITLE
Print name of created cache archive

### PR DESCRIPTION
**Pull request by cotsog**

### CODE REVIEW

File: bin/casher [hunk 1]
```diff
@@ -130,10 +130,10 @@ class Casher
       return
     end
 
-    msg "changes detected, packing new archive"
-
     @push_tar  = File.expand_path('push.tgz', @casher_dir) if path_ext(url) == 'tgz'
 
+    msg "changes detected, packing new archive: #{@push_tar}"
```
Comment (BanzaiMan) [inline bin/casher]: `@push_tar` is going to be a simple `push.tbz`, and it'll be uninteresting.

If you are interested in the name, you'd want to display parts of `url`.

Comment (cotsog) [reply bin/casher]: @BanzaiMan Should I just change `@push_tar` for `url`? e.g.

```ruby
msg "changes detected, packing new archive: #{url}"
```

Thanks!

Comment (BanzaiMan) [reply bin/casher]: `url` would include the signed URL for `POST`ing the archive, so we should not reveal it. We'll have to drop the request parameters (everything after the last `&`), as well as the parts leading up to (and including) the penultimate `/`.

**Comment by cotsog** [comment]
Example run => https://travis-ci.com/travis-pro/test-project/builds/88304759#L472-L474

File: bin/casher [hunk 2]
```diff
@@ -130,18 +130,20 @@ class Casher
...
+
     tar(:c, @push_tar, *cached_directories) do
       @counter += 1
       puts "." if @counter % 5 == 0
       sleep 1
     end
 
-    msg "uploading archive"
-    unless system "curl -T %p %p -f -v >#{@casher_dir}/push.log 2>#{@casher_dir}/push.err.log" % [@push_tar, url]
+    msg "uploading #{%r(([^/]+?\/[^/]+?)(\?.*)?\?).match(url)[1]}"
```
Comment (BanzaiMan) [inline bin/casher]: Getting better! Let's extract `%r(([^/]+?\/[^/]+?)(\?.*)?\?).match(url)[1]` to a method, so we don't have to repeat it here.

**Comment by BanzaiMan** [comment]
https://github.com/travis-ci/casher/commit/5971c448b805987a615962dc6f6b46ffedca4de8 is the bash equivalent.

**Comment by BanzaiMan** [comment]
This has been deployed to production.

**Pull request merged**